### ISSUE TITLE
Add IPv6 support

### ISSUE THREAD

**Issue by giomasce** [cms-dev/cms#157]
At the moment CMS supports only IPv4, but the Internet is (very slowly) moving towards a new protocol: IPv6! Do we really want to stay behind?

**Comment by lerks** [comment]
The only outstanding issue for IPv6 support that appears here is being addressed in #738. I believe we can close this after that PR is merged.

**Comment by edomora97** [comment]
IPv6 doesn't really work in a network IPv6 only. If the services are accessible only via IPv6 there are many errors in the RPC.
The problem we noticed is caused by https://github.com/cms-dev/cms/blob/bfb359ba21935785b47ecb21b86ff4fcdb1f3522/cms/io/rpc.py#L481

We patched cms to make it work during the night, maybe a better fix should be implemented soon, for now here it's the diff from https://github.com/cms-dev/cms/commit/279c0fd9acfe16bf004267b7d89ceb9c3a0190cd

```diff
diff --git a/cms/io/rpc.py b/cms/io/rpc.py
index 9ec358b0..a8d42217 100644
--- a/cms/io/rpc.py
+++ b/cms/io/rpc.py
@@ -20,6 +20,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import functools
+import ipaddress
 import json
 import logging
 import socket
@@ -478,7 +479,9 @@ class RemoteServiceClient(RemoteServiceBase):
 
         """
         try:
-            sock = gevent.socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            info = socket.getaddrinfo(self.remote_address.ip, self.remote_address.port)
+            info = info[0]
+            sock = gevent.socket.socket(info[0], socket.SOCK_STREAM)
             sock.connect(self.remote_address)
         except OSError as error:
             logger.debug("Couldn't connect to %s: %s.",
diff --git a/cms/io/service.py b/cms/io/service.py
index 942319e3..63be02d3 100644
--- a/cms/io/service.py
+++ b/cms/io/service.py
@@ -26,6 +26,7 @@ using gevent and JSON encoding.
 
 import errno
 import functools
+import ipaddress
 import logging
 import os
 import pwd
@@ -97,7 +98,8 @@ class Service:
                               "Is it specified in core_services in cms.conf?" %
                               (self._my_coord,))
 
-        self.rpc_server = StreamServer(address, self._connection_handler)
+        info = socket.getaddrinfo(address.ip, address.port)
+        self.rpc_server = StreamServer(info[0][-1][:2], self._connection_handler)
         self.backdoor = None
 
     def initialize_logging(self):
@@ -158,8 +160,7 @@ class Service:
 
         """
         try:
-            ipaddr, port = address
-            ipaddr = gevent.socket.gethostbyname(ipaddr)
+            ipaddr, port = address[:2]
             address = Address(ipaddr, port)
         except OSError:
             logger.warning("Unexpected error.", exc_info=True)
```

Not sure if it is really enough, if it works on py2, if we're doing the right thing. It's an hack we developed late in the night, right now we don't have much time to work on it but at least it's a start :stuck_out_tongue:
